src/models/Level.php

```diff
@@ -625,9 +625,47 @@ private static function levelFromRow(Map<string, string> $row): Level {
     $level = await self::gen($level_id);
     await Country::genSetUsed($level->getEntityId(), false);
 
+    // Remove team points for level
+    $scores = await ScoreLog::genAllScoresByLevel($level_id);
+    foreach ($scores as $score) {
+      $team_id = $score->getTeamId();
+      $points = $score->getPoints();
+      await $db->queryf(
+        'UPDATE teams SET points = points - %d WHERE id = %d',
+        $points,
+        $team_id,
+      );
+    }
+
+    // Remove hint penalties from teams points for level
+    $hints = await HintLog::genAllHintsByLevel($level_id);
+    foreach ($hints as $hint) {
+      $team_id = $hint->getTeamId();
+      $penalty = $hint->getPenalty();
+      await $db->queryf(
+        'UPDATE teams SET points = points + %d WHERE id = %d',
+        $penalty,
+        $team_id,
+      );
+    }
+
+    // Delete all references to level
     await $db->queryf('DELETE FROM levels WHERE id = %d LIMIT 1', $level_id);
+    await $db->queryf('DELETE FROM hints_log WHERE level_id = %d', $level_id);
```

Comment:
Remember to try and combine awaits together with something like `\HH\Asio\va`

You can do that for loops too by putting the `Awaitables` in an array.